File: scripts/test_mcp_client.js
#!/usr/bin/env node
/**
 * Minimal MCP client testing through SentinelGate proxy — Session A6.
 */
const PROXY_URL = process.env.SG_PROXY_URL || "http://localhost:8080/mcp";
const API_KEY = process.env.SG_API_KEY || "";

let passed = 0;
let failed = 0;
let reqId = 0;

async function mcpCall(method, params = {}) {
  reqId++;
  try {
    const resp = await fetch(PROXY_URL, {
      method: "POST",
      headers: {
        "Content-Type": "application/json",
        "Authorization": `Bearer ${API_KEY}`
      },
      body: JSON.stringify({ jsonrpc: "2.0", id: reqId, method, params }),
      signal: AbortSignal.timeout(30000)
    });
    return await resp.json();
  } catch (e) {
    return { error: String(e) };
  }
}

function check(testName, condition, detail = "") {
  if (condition) {
    passed++;
    console.log(`PASS:${testName}:${detail}`);
  } else {
    failed++;
    console.log(`FAIL:${testName}:${detail}`);
  }
}

async function main() {
  if (!API_KEY) {
    console.log("FAIL:setup:SG_API_KEY environment variable required");
    process.exit(1);
  }

  // Test 1: Initialize
  let result = await mcpCall("initialize", {
    protocolVersion: "2024-11-05",
    capabilities: {},
    clientInfo: { name: "nodejs-test-agent", version: "1.0" }
  });
  const hasServerInfo = result?.result?.serverInfo != null;
  check("initialize", hasServerInfo,
    result?.result?.serverInfo?.name || "?");

  // Test 2: List tools
  result = await mcpCall("tools/list");
  const tools = result?.result?.tools || [];
  check("tools_list", tools.length > 0, `${tools.length} tools`);
  tools.slice(0, 5).forEach(t =>
    console.log(`  TOOL: ${t.name}: ${(t.description || "").slice(0, 60)}`)
  );

  // Test 3: Call echo tool
  result = await mcpCall("tools/call", {
    name: "echo",
    arguments: { message: "Hello from Node.js agent!" }
  });
  let contentText = result?.result?.content?.[0]?.text || "";
  check("tool_call_echo", result?.result != null && contentText.includes("Hello"),
    contentText.slice(0, 80));

  // Test 4: Call get-sum tool
  result = await mcpCall("tools/call", {
    name: "get-sum",
    arguments: { a: 7, b: 13 }
  });
  contentText = result?.result?.content?.[0]?.text || "";
  check("tool_call_get_sum", result?.result != null && contentText.includes("20"),
    contentText.slice(0, 80));

  // Summary
  console.log(`SUMMARY:${passed}:${failed}`);
  process.exit(failed === 0 ? 0 : 1);
}

main().catch(e => {
  console.log(`FAIL:uncaught:${e.message}`);
  process.exit(1);
});

File: scripts/test_mcp_client.py
#!/usr/bin/env python3
"""Minimal MCP client testing through SentinelGate proxy — Session A6."""
import json
import os
import sys
import urllib.request
import urllib.error

PROXY_URL = os.environ.get("SG_PROXY_URL", "http://localhost:8080/mcp")
API_KEY = os.environ.get("SG_API_KEY", "")

passed = 0
failed = 0
req_id = 0


def mcp_call(method, params=None):
    global req_id
    req_id += 1
    payload = json.dumps({
        "jsonrpc": "2.0",
        "id": req_id,
        "method": method,
        "params": params or {}
    }).encode()
    req = urllib.request.Request(
        PROXY_URL,
        data=payload,
        headers={
            "Content-Type": "application/json",
            "Authorization": f"Bearer {API_KEY}"
        }
    )
    try:
        with urllib.request.urlopen(req, timeout=30) as resp:
            return json.loads(resp.read())
    except urllib.error.HTTPError as e:
        return {"error": f"HTTP {e.code}", "body": e.read().decode()[:200]}
    except Exception as e:
        return {"error": str(e)}


def check(test_name, condition, detail=""):
    global passed, failed
    if condition:
        passed += 1
        print(f"PASS:{test_name}:{detail}")
    else:
        failed += 1
        print(f"FAIL:{test_name}:{detail}")


def main():
    if not API_KEY:
        print("FAIL:setup:SG_API_KEY environment variable required")
        sys.exit(1)

    # Test 1: Initialize
    result = mcp_call("initialize", {
        "protocolVersion": "2024-11-05",
        "capabilities": {},
        "clientInfo": {"name": "python-test-agent", "version": "1.0"}
    })
    has_server_info = "result" in result and "serverInfo" in result.get("result", {})
    check("initialize", has_server_info,
          result.get("result", {}).get("serverInfo", {}).get("name", "?"))

    # Test 2: List tools
    result = mcp_call("tools/list")
    tools = result.get("result", {}).get("tools", [])
    check("tools_list", len(tools) > 0, f"{len(tools)} tools")
    for t in tools[:5]:
        print(f"  TOOL: {t['name']}: {t.get('description', '')[:60]}")

    # Test 3: Call echo tool
    result = mcp_call("tools/call", {
        "name": "echo",
        "arguments": {"message": "Hello from Python agent!"}
    })
    has_result = "result" in result
    content_text = ""
    if has_result:
        contents = result["result"].get("content", [])
        if contents:
            content_text = contents[0].get("text", "")
    check("tool_call_echo", has_result and "Hello" in content_text, content_text[:80])

    # Test 4: Call get-sum tool
    result = mcp_call("tools/call", {
        "name": "get-sum",
        "arguments": {"a": 7, "b": 13}
    })
    has_result = "result" in result
    content_text = ""
    if has_result:
        contents = result["result"].get("content", [])
        if contents:
            content_text = contents[0].get("text", "")
    check("tool_call_get_sum", has_result and "20" in content_text, content_text[:80])

    # Summary
    print(f"SUMMARY:{passed}:{failed}")
    sys.exit(0 if failed == 0 else 1)


if __name__ == "__main__":
    main()
